pkg/db/incident.go

```diff
@@ -62,36 +64,62 @@ func (i *Incident) GetIncidentUpdates() *apiServerDefinition.IncrementalList {
 	return &updates
 }
 
+func isMaintenance(impacts *[]Impact) bool {
+	if impacts == nil {
+		return false
+	}
+
+	for _, impact := range *impacts {
+		if impact.Severity == nil {
+			continue
+		}
+
+		if *impact.Severity == api.MaintenanceSeverity {
+			return true
+		}
+	}
+
+	return false
+}
```

Comment:
The way this is written, any incident containing even a single impact marked with the maintenance severity value will be classified as a maintenance event, even if it were to contain, let's say, 50 non-maintenance impacts. Is this the correct behaviour? Can this even occur?

Reply:
This can happen. The API does not prohibit the creation of incidents having mixed severities.

For example an "normal" incident on network speed might effect the database with a lot of data throughput more, then some small API running somewhere. Both are affected, but the DB impact may be a 90 severity and the API impact just a 30. 

As maintenance windows are just incidents with an impact of severity 0, there is nothing stopping a user from creating an incident with multiple impacts, of which 0..n could be maintenance severity. 

If a maintenance severity is used, this ensures at least the creation of an end time, to qualify it as a "window".

Per API spec this behavior is not prevented, it's not logically to use and has no use case in my opinion, but it is possible.